src/service/reporting/reportService.ts

```diff
@@ -64,14 +64,26 @@ Promise<{ csvFile: JobsFile }> => {
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-expect-error
   const emailIds = (await tableService.getJobData('skillsSync', 'emailIds')).split(',')
-  let csvFilenamePrefix
+  let csvFilenamePrefix: string
+  const lastFile = {
+    lastFileOperation: await tableService.getJobData('skillsSync', 'lastFileOperation'),
+    lastFileDate: await tableService.getJobData('skillsSync', 'lastFileDate'),
+    lastFileSequenceNumber: await tableService.getJobData('skillsSync', 'lastFileSequenceNumber')
+  }
+  const formatTokens = 'DDMMYYYY'
+  const formattedDate = dayjs().format(formatTokens)
+  const sequenceNumber = ((lastFile.lastFileSequenceNumber != null) && (lastFile.lastFileDate != null) && formattedDate === lastFile.lastFileDate) ? parseInt(lastFile.lastFileSequenceNumber) + 1 : 1
+  let operation
   if (lastSuccessTimestamp === undefined) {
     csvFilenamePrefix = config.jobs.skillsCompletedLearnerRecords.csvFilenamePrefixCreate
+    operation = 'create'
   } else {
     csvFilenamePrefix = config.jobs.skillsCompletedLearnerRecords.csvFilenamePrefixUpdate
+    operation = 'update'
   }
-  const csvFileName = getCurrentDateFileName(csvFilenamePrefix) + '.csv'
+  const csvFileName = `${csvFilenamePrefix}_${formattedDate}_${sequenceNumber}.csv`
```

Comment:
I think we can split the filename generation logic into its own function, eg `getFullFilename()`.

Reply:
Done